tests/acceptance/features/bootstrap/Provisioning.php

```diff
@@ -3120,58 +3120,36 @@ public function userTriesToSendGroupCreationRequestUsingTheAPI(string $user, str
 	 */
 	public function createTheGroup(string $group, ?string $method = null):void {
 		//guess yourself
-		if ($method === null) {
-			if ($this->isTestingWithLdap()) {
-				$method = "ldap";
-			} elseif (OcisHelper::isTestingWithGraphApi()) {
-				$method = "graph";
-			} else {
-				$method = "api";
-			}
-		}
 		$group = \trim($group);
-		$method = \trim(\strtolower($method));
 		$groupCanBeDeleted = false;
 		$groupId = null;
-		switch ($method) {
-			case "api":
-				$result = UserHelper::createGroup(
-					$this->getBaseUrl(),
-					$group,
-					$this->getAdminUsername(),
-					$this->getAdminPassword(),
-					$this->getStepLineRef()
-				);
-				if ($result->getStatusCode() === 200) {
-					$groupCanBeDeleted = true;
-				} else {
-					throw new Exception(
-						"could not create group '$group'. "
-						. $result->getStatusCode() . " " . $result->getBody()
-					);
-				}
-				break;
-			case "ldap":
+		if ($method === null) {
+			if ($this->isTestingWithLdap()) {
 				try {
 					$this->createLdapGroup($group);
 				} catch (LdapException $e) {
 					throw new Exception(
 						"could not create group '$group'. Error: $e"
 					);
 				}
-				break;
-			case "graph":
-				$newGroup = $this->graphContext->adminHasCreatedGroupUsingTheGraphApi($group);
+			} elseif (OcisHelper::isTestingWithGraphApi()) {
+				$newGroup = $this->graphContext->createGroup($group);
+				if ($newGroup->getStatusCode() === 201) {
```

Comment:
don't we have a method to check status code rather than explicitly checking the status code?

Reply:
yes and I don't think we should use the status check in the function itself that should be the part of given/then step?